libcudacxx/include/cuda/std/__cccl/compiler.h

```diff
@@ -56,7 +56,14 @@
 #  define _CCCL_CUDA_COMPILER
 #endif // cuda compiler available
 
-#if defined(__CUDACC__) || defined(_CCCL_CUDA_COMPILER_NVHPC)
+// clang-cuda does not define __CUDACC_VER_MAJOR__ and friends
+#if defined(_CCCL_CUDA_COMPILER_CLANG)
+#  define _CCCL_CUDACC
+#  define _CCCL_CUDACC_VER_MAJOR CUDA_VERSION / 1000
+#  define _CCCL_CUDACC_VER_MINOR (CUDA_VERSION % 1000) / 10
+#  define _CCCL_CUDACC_VER_BUILD 0
+#  define _CCCL_CUDACC_VER       CUDA_VERSION * 100
+#elif defined(_CCCL_CUDA_COMPILER)
```

Comment:
@gevtushenko We should check whether the macros are defined in `cuda_runtime_api.h`

Reply:
Those macros are explicitly defined by nvcc and not a header https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#nvcc-identification-macro